include/cadmium/concept/coupled_model_assert.hpp

```diff
@@ -37,7 +37,7 @@ namespace cadmium {
     namespace concept {
         //forward declaration
         template<typename MODEL>
-        constexpr void coupled_model_float_time_assert();
+        constexpr void pdevs_coupled_model_float_time_assert();
```

Comment:
1. I would rater use namespaces concept::pdevs::couped_model_float_time_assert() instead of adding the prefix "pdevs_" in the name. This worth for all the code and not just this name. 

2. I would put all the common to all the formalism (pdevs and devs) code in common files but for the specific code of each formalism I would use separated files to avoid user compiling unused code of other  formalism.

Reply:
1. In particular for coupled_mode_float_time_assert I would also add a "impl" namespace since it is not supose to be used, but I will make it as you suggest for now and make that improvement later, we have a Ticket to do that in many other places already.
2. Unused templates code is not actually compiled, so it will not make much difference. Lets discuss it further. I'm worried about have to add code because of the change since common may end up above and below the specific formalism code and there is no forward declaration that could be used for this.